pkg/api/customization/cluster/actions_etcd.go

```diff
@@ -38,7 +40,10 @@ func (a ActionHandler) BackupEtcdHandler(actionName string, action *types.Action
 		return errors.Wrapf(err, "failed to get Cluster by ID %s", apiContext.ID)
 	}
 
-	newBackup := etcdbackup.NewBackupObject(cluster, true)
+	newBackup, err := etcdbackup.NewBackupObject(cluster, true)
+	if err != nil {
+		return err
```

Comment:
We should also return a response for the action using apiContext.WriteResponse  and return error using errors.Wrapf with a message that can help debugging

Reply:
yup ill add